contracts/token/ERC721/ERC721BasicToken.sol

```diff
@@ -0,0 +1,284 @@
+   *  is an operator of the owner, or is the owner of the token
+   */
+  function isApprovedOrOwner(address _spender, uint256 _tokenId) internal view returns (bool) {
+    address owner = ownerOf(_tokenId);
+    return _spender == owner || getApproved(_tokenId) == _spender || isApprovedForAll(owner, _spender);
+  }
+
+  /**
+  * @dev Internal function to mint a new token
+  * @dev Reverts if the given token ID already exists
+  * @param _to The address that will own the minted token
+  * @param _tokenId uint256 ID of the token to be minted by the msg.sender
+  */
+  function doMint(address _to, uint256 _tokenId) internal {
+    require(_to != address(0));
+    addToken(_to, _tokenId);
+    totalTokens = totalTokens.add(1);
+    Transfer(0x0, _to, _tokenId);
+  }
+
+  /**
+  * @dev Internal function to burn a specific token
+  * @dev Reverts if the token does not exist
+  * @param _tokenId uint256 ID of the token being burned by the msg.sender
+  */
+  function doBurn(uint256 _tokenId) onlyOwnerOf(_tokenId) internal {
+    clearApproval(msg.sender, _tokenId);
+    removeToken(msg.sender, _tokenId);
+    totalTokens = totalTokens.sub(1);
+    Transfer(msg.sender, 0x0, _tokenId);
+  }
+
+  /**
+  * @dev Internal function to clear current approval and transfer the ownership of a given token ID
+  * @param _from address which you want to send tokens from
+  * @param _to address which you want to transfer the token to
+  * @param _tokenId uint256 ID of the token to be transferred
+  * @param _data bytes data to send along with a safe transfer check
+  * @param _safe bool whether to perform a safe transfer
+  */
+  function clearApprovalAndTransfer(address _from, address _to, uint256 _tokenId, bytes _data, bool _safe) internal {
+    require(_from != address(0));
+    require(_to != address(0));
+    require(_to != ownerOf(_tokenId));
+
+    clearApproval(_from, _tokenId);
+    removeToken(_from, _tokenId);
+    addToken(_to, _tokenId);
+    
+    require(!_safe || checkAndCallSafeTransfer(_from, _to, _tokenId, _data));
+
+    Transfer(_from, _to, _tokenId);
+  }
+
+  /**
+  * @dev Internal function to clear current approval of a given token ID
+  * @dev Reverts if the given address is not indeed the owner of the token
+  * @param _owner owner of the token
+  * @param _tokenId uint256 ID of the token to be transferred
+  */
+  function clearApproval(address _owner, uint256 _tokenId) internal {
+    require(ownerOf(_tokenId) == _owner);
+    if (tokenApprovals[_tokenId] != 0) {
+      tokenApprovals[_tokenId] = 0;
+      Approval(_owner, 0, _tokenId);
+    }
+  }
+
+  /**
+  * @dev Internal function to add a token ID to the list of a given address
+  * @param _to address representing the new owner of the given token ID
+  * @param _tokenId uint256 ID of the token to be added to the tokens list of the given address
+  */
+  function addToken(address _to, uint256 _tokenId) internal {
```

Comment:
Consider making it `private` rather than `internal` so that every addition is _forced_ to be made via `doMint`. In this way we also make sure that the `totalTokens`count is always updated  and that the corresponding event `Transfer` is transmitted.

Reply:
Note that the `ERC721Token.sol` full implementation would break if we make this method private, since the `ownedTokens` fields need to be updated there.